### ISSUE TITLE
no-unused-keys reports false positives when a used key is a prefix of another used key

### ISSUE THREAD

**Issue by iichernyshev** [intlify/eslint-plugin-vue-i18n#737]
**Tell us about your environment**

- **ESLint version:** 9.39.2
- **eslint-plugin-vue-i18n version:** 4.4.0
- **Node version:** 22.13.0

**Please show your full configuration:**

```js
import vueI18n from '@intlify/eslint-plugin-vue-i18n'
import jsoncParser from 'jsonc-eslint-parser'
import vueParser from 'vue-eslint-parser'

export default [
  {
    plugins: {
      '@intlify/vue-i18n': vueI18n,
    },
    settings: {
      'vue-i18n': {
        localeDir: './src/locales/*.json',
      },
    },
  },
  {
    files: ['src/**/*.vue'],
    languageOptions: {
      parser: vueParser,
      parserOptions: {
        ecmaVersion: 'latest',
        sourceType: 'module',
      },
    },
  },
  {
    files: ['src/locales/*.json'],
    languageOptions: {
      parser: jsoncParser,
    },
    rules: {
      '@intlify/vue-i18n/no-unused-keys': ['error', {
        src: './src',
        extensions: ['.vue'],
      }],
    },
  },
]
```

**What did you do?**
I use a Vue 3 SFC with a local `<i18n locale="en-GB">` block and a global JSON locale file.

```vue
<script setup>
import { useI18n } from 'vue-i18n'

const { t } = useI18n()
</script>

<template>
  <main>
    <button type="button">
      {{ t('action.close') }}
    </button>

    <p>
      {{ t('action') }}
    </p>
  </main>
</template>

<i18n locale="en-GB">
{
  "action": "Local action label"
}
</i18n>
```

```js
{
  "action": {
    "close": "Close"
  }
}
```

**What did you expect to happen?**
`action.close` should not be reported as unused because it is used in App.vue. The local SFC key `action` should not make the global JSON key `action.close` appear unused.

**What actually happened?**

```text
src/locales/en-GB.json
  3:5  error  unused 'action.close' key  @intlify/vue-i18n/no-unused-keys

✖ 1 problem (1 error, 0 warnings)
```
It looks like `getUsedKeysMap` overwrites a previously built nested branch when a shorter key is processed later.

**Repository to reproduce this issue**
https://github.com/iichernyshev/no-unused-keys-example

I have a possible fix here:

https://github.com/iichernyshev/eslint-plugin-vue-i18n/commit/4e28f50

The fix preserves an existing branch instead of overwriting it
